sklearn/decomposition/tests/test_sparse_pca.py

```diff
@@ -86,6 +91,36 @@ def test_fit_transform(norm_comp):
                          Y, ridge_alpha=None)
 
 
+@pytest.mark.filterwarnings("ignore:normalize_components")
+@pytest.mark.parametrize("norm_comp", [False, True])
+def test_fit_transform_variance(norm_comp):
+    # This function asserts that the variance computed by SparsePCA is the
+    # same as the variance in PCA when the components are orthogonal.
+    alpha = 1
+    rng = np.random.RandomState(0)
+    X, _, _ = generate_toy_data(3, 10, (8, 8), random_state=rng)  # wide array
+    # init spca and pca
+    spca_lars = SparsePCA(n_components=3, method='lars', alpha=alpha,
+                          random_state=0, normalize_components=norm_comp)
+    spca_lars.fit(X)
+    assert spca_lars.explained_variance_.shape == (3,)
+    pca = PCA(n_components=3, random_state=0)
+
+    # fit PCA
+    pca.fit(X)
+
+    explained_variance = pca.explained_variance_
+
+    # force the components in spca_lars to be the same as in pca
+    spca_lars.components_ = pca.components_
```

Comment:
How about a string version of that test that first applies QR-decomposition on spca_lars.components_ and then computes and compares the explained variances against PCA's output?

Reply:
I assume by that you mean something like this:
```py3
# Force matrix to be orthogonal. sparse PCA must then be the same as PCA.
   X_orthogonal, R = np.linalg.qr(X)
   spca_lars.fix(X_orthogonal)
   pca.fit(X_orthogonal)

   explained_variance = pca.explained_variance_
   explained_variance_sparse = _get_explained_variance(
      X, spca_lars.components_)
```

I would have assumed this to be right. I will not add it. I don't have time to update it (I'm working in a completely different field). I suggest to either:
- make correction here
- add commit to this PR
- we can close it and someone else continue it
- make an issue to add this test

What do you think? I apologize for my latency. It is very hard for me to step into the open source world right now.

Reply:
Yep, this goes into the direction I had in mind. Would be a nice option for a more careful test if feasible.